azure-pipelines.yml

```diff
@@ -193,6 +193,7 @@ stages:
             mkdir -p /tmp/terra-tests
             cp -r test /tmp/terra-tests/.
             cp .stestr.conf /tmp/terra-tests/.
+            cp -r .stestr /tmp/terra-tests/. || :
```

Comment:
Here we explicitly throw away any error in `cp`, but later on down the file, we assume that `.stestr` will exist in `/tmp/terra-tests`.  I'm assuming that the later case should be safe even if this line fails, because `stestr` will create the directory if it hasn't been restored from the cache?

Reply:
Right, that's what my intent here was. `stestr` will create it's local repository directory when `stestr run` is invoked in `/tmp/terra-tests` regardless of whether we copy it from the cache or not. So I was trying to make the script robust against `.stestr` not being pulled from azure's cache here.

Reply:
Perfect, that's how I read it too.